docs/api/vi.md

```diff
@@ -1,409 +1,432 @@
+---
+outline: deep
+---
+
 # Vi
 
-Vitest provides utility functions to help you out through its `vi` helper. You can access it globally (when [globals configuration](/config/#globals) is **enabled**), or import from `vitest`:
+Vitest provides utility functions to help you out through its `vi` helper. You can access it globally (when [globals configuration](/config/#globals) is enabled), or import it from `vitest` directly:
 
 ```js
 import { vi } from 'vitest'
 ```
 
-## vi.advanceTimersByTime
-
-- **Type:** `(ms: number) => Vitest`
-
-  Works just like `runAllTimers`, but will end after passed milliseconds. For example this will log `1, 2, 3` and will not throw:
-
-  ```ts
-  let i = 0
-  setInterval(() => console.log(++i), 50)
+## Mock Modules
 
-  vi.advanceTimersByTime(150)
-  ```
+This section describes API that you can use when [mocking a module](/guide/mocking#modules). Beware that Vitest doesn't support mocking modules imported using `require()`.
 
-### vi.advanceTimersByTimeAsync
-
-- **Type:** `(ms: number) => Promise<Vitest>`
+### vi.mock
 
-  Works just like `runAllTimersAsync`, but will end after passed milliseconds. This will include asynchronously set timers. For example this will log `1, 2, 3` and will not throw:
+- **Type**: `(path: string, factory?: (importOriginal: () => unknown) => unknown) => void`
 
-  ```ts
-  let i = 0
-  setInterval(() => Promise.resolve().then(() => console.log(++i)), 50)
+Substitutes all imported modules from provided `path` with another module. You can use configured Vite aliases inside a path. The call to `vi.mock` is hoisted, so it doesn't matter where you call it. It will always be executed before all imports. If you need to reference some variables outside of its scope, you can define them inside [`vi.hoisted`](/api/vi#vi-hoisted) and reference them inside `vi.mock`.
 
-  await vi.advanceTimersByTimeAsync(150)
-  ```
+::: warning
+`vi.mock` works only for modules that were imported with the `import` keyword. It doesn't work with `require`.
 
-## vi.advanceTimersToNextTimer
+Vitest statically analyzes your files to hoist `vi.mock`. It means that you cannot use `vi` that was not imported directly from `vitest` package (for example, from some utility file). To fix this, always use `vi.mock` with `vi` imported from `vitest`, or enable [`globals`](/config/#globals) config option.
 
-- **Type:** `() => Vitest`
+Vitest will not mock modules that were imported inside a [setup file](/config/#setupfiles) because they are cached by the time a test file is running. You can call [`vi.resetModules()`](#vi-resetmodules) inside [`vi.hoisted`](#vi-hoisted) to clear all modules caches before running a test file.
```

Comment:
```suggestion
Vitest will not mock modules that were imported inside a [setup file](/config/#setupfiles) because they are cached by the time a test file is running. You can call [`vi.resetModules()`](#vi-resetmodules) inside [`vi.hoisted`](#vi-hoisted) to clear all module caches before running a test file.
```